### DISCUSSION TITLE
How to translate zed Ui?

### DISCUSSION THREAD

**Discussion by Angsimosaurus** [zed-industries/zed#27609] [Feature Requests]
### Check for existing issues

- [x] Completed

### Describe the feature

Hello, zed developers! I couldn't wait for the official Windows version of zed, so I've already completed building the current Windows version, but it seems to only support English. I'm planning to complete the Korean translation before the official Windows version of zed is released and continue to update it. Is there any translation guide available?

Additionally, I have one suggestion: I would like the program to maintain the open tabs and their contents even when closed and reopened, similar to notepad++ or sublime text.

Oh! We also need an option to make only one instance of the program run!

Thank you for this wonderful open-source program!

### If applicable, add mockups / screenshots to help present your vision of the feature

![008](https://github.com/user-attachments/assets/2eeb75cf-f42e-4f84-9888-51d29fc96054)
![image](https://github.com/user-attachments/assets/db242b66-7b10-45cf-abf8-e40032978037)
![image](https://github.com/user-attachments/assets/97fa0567-f337-4725-a005-833767e8be38)

**Comment by xnousnow** [comment]
I'm also a Korean user and I'd love to help with the translation if possible!

Also check out these: #7409, [@nriver/zed-translation](https://github.com/Nriver/zed-translation)

By the way, I think it would be helpful to separate the different features you mentioned into individual issues rather than combining them all into one. This way, each suggestion can be addressed more effectively.

**Comment by yeskunall** [comment]
@Angsimosaurus -- you’ve coupled in a few issues into one discussion, however --

> Additionally, I have one suggestion: I would like the program to maintain the open tabs and their contents even when closed and reopened, similar to notepad++ or sublime text.

I believe you’re looking for the `restore_on_file_reopen` setting.

> Oh! We also need an option to make only one instance of the program run!

This is now the default!

But this discussion is mostly about i18n support in Zed, for which we already have a long outstanding [feature issue](https://github.com/zed-industries/zed/issues/7409) open, so I’m going to close this one in favor of that.
